### PR TITLE
[APM] Convert `_.get` and `ts-optchain` to idx

### PR DESCRIPTION
**Pull request by sqren**
Blocked by https://github.com/elastic/kibana/pull/28068

### CODE REVIEW

File: x-pack/plugins/apm/server/lib/services/get_services.ts [hunk 1]
```diff
@@ -108,23 +108,23 @@ export async function getServices(
...
+    const totalErrors = idx(errors, _ => _.doc_count) || 0;
 
     const deltaAsMinutes = (end - start) / 1000 / 60;
     const transactionsPerMinute = totalTransactions / deltaAsMinutes;
     const errorsPerMinute = totalErrors / deltaAsMinutes;
 
     return {
       serviceName: bucket.key,
-      agentName: oc(bucket).agents.buckets[0].key(),
+      agentName: idx(bucket, _ => _.agents.buckets[0].key) || undefined, // TODO: get rid of unneeded undefined
```
Comment (ogupte) [inline x-pack/plugins/apm/server/lib/services/get_services.ts]: TODO to get rid of the `undefined`

File: x-pack/plugins/apm/public/components/shared/Stacktrace/FrameHeading.tsx [hunk 2]
```diff
@@ -32,7 +32,7 @@ const FrameHeading: React.SFC<Props> = ({ stackframe, isLibraryFrame }) => {
   const FileDetail = isLibraryFrame
     ? LibraryFrameFileDetail
     : AppFrameFileDetail;
-  const lineNumber: number = get(stackframe, 'line.number');
+  const lineNumber = idx(stackframe, _ => _.line.number) || 0;
```
[the remark this answers is not in the document]
Comment (sqren) [reply x-pack/plugins/apm/public/components/shared/Stacktrace/FrameHeading.tsx]: I was getting errors on line 40 `{lineNumber > 0 && (`

[the remark this answers is not in the document]
Comment (sqren) [reply x-pack/plugins/apm/public/components/shared/Stacktrace/FrameHeading.tsx]: I could also change it back to: `const lineNumber: number = idx(stackframe, _ => _.line.number)` (I think)

File: x-pack/plugins/apm/server/lib/services/get_services.ts (hunk 1 above)
Comment (sqren) [reply x-pack/plugins/apm/server/lib/services/get_services.ts]: Hmm... I wonder why I ever added that...

**Comment by sqren** [comment]
Closed in favour of https://github.com/elastic/kibana/pull/29297.

**Pull request closed without merging**